presentation/src/main/java/com/smile/presentation/MainActivityFragment.kt

```diff
@@ -58,7 +60,7 @@ class MainActivityFragment : BaseFragment() {
 
     private fun populateData() {
         viewModel.characterLiveData.observe(this, Observer {
-            characterItems.addAll(it)
+            viewModel.characterItems.addAll(it)
```

Comment:
I don't think its best for UI to change data in viewmodel. It is meant to only observe and send out actions.